samcli/lib/init/cli_template_modifier.py

```diff
@@ -0,0 +1,189 @@
+            ]
+            self.template = (
+                self.template[:resource_section_position] + global_section + self.template[resource_section_position:]
+            )
+
+    def section_position(self, section: str, position: int = 0) -> int:
+        """
+        validate if a section in the template exist
+
+        Parameters
+        ----------
+        section : str
+            A section in the SAM template
+        position : int
+            position to start searching for the section
+
+        Returns
+        -------
+        int
+            index of section in the template list
+        """
+        template = self.template[position:]
+        for index, line in enumerate(template):
+            if line.startswith(section):
+                section_index = index + position if position else index
+                return section_index
+        return -1
+
+    def add_fields_to_section(self, position: int, fields: str) -> Any:
+        """
+        Adds fields to section in the template
+
+        Parameters
+        ----------
+        position : int
+            position to start searching for the section
+        fields : str
+            fields to be added to the SAM template
+
+        Returns
+        -------
+        list
+            array with updated template data
+        """
+        template = self.template[position:]
+        for index, line in enumerate(template):
+            if not (line.startswith(" ") or line.startswith("#")):
+                return self.template[: position + index] + fields + self.template[position + index :]
+        return self.template
+
+    def field_position(self, position, field) -> Any:
+        """
+        Checks if the field needed to be added to the SAM template already exist in the template
+
+        Parameters
+        ----------
+        position : int
+            section position to start the search
+        field : str
+            Field name
+
+        Returns
+        -------
+        int
+            index of the field if it exist else -1
+        """
+        template = self.template[position:]
+        for index, line in enumerate(template):
+            if field in line:
+                return position + index
+            if not (line.startswith(" ") or line.startswith("#")):
+                break
+        return -1
+
+    def sanity_check(self) -> bool:
+        """
+        Conducts sanity check on template using yaml parser to ensure the updated template meets
+        CFN template criteria
+
+        Returns
+        -------
+        bool
+            True if templates passes sanity check else False
+        """
+        try:
+            parse_template = parse_yaml_file(self.template_location)
+            return bool(parse_template)
+        except ParserError:
+            link = "https://docs.aws.amazon.com/lambda/latest/dg/services-xray.html"
+            message = f"Error: Unable to add Tracing to the project. To learn more Tracing visit {link}"
```

Comment:
It should be 
> ... to learn more **about** Tracing

And potentially you can get an overall better message from UX or PM